### DISCUSSION TITLE
Help regarding custom payment provider

### DISCUSSION THREAD

**Discussion by TALHA0713** [medusajs/medusa#10787] [Q&A]
HI everyone,
I want to create a payment provider and i have integrate medusa v1 docs can i shift to V2 instance withou facing any issue.
Also i want to create add online payment method which works in Pakistan so for this i want to use the jazz cash or any other payment gateway but i dont know how i can do this 
for that after reasearch i am found this [](https://docs.medusajs.com/v1/modules/carts-and-checkout/backend/add-payment-provider)
and am add in my code but I did not understand how i can add my payment gateway attach in my cart i need some suggestion and proper flow
Please give me some support 
[](https://docs.medusajs.com/v1/modules/carts-and-checkout/backend/add-payment-provider ) this code is write in backend admin code or store front code also ???
I want to ask some question also 
mt559310@gmail.com thats my email please contact me i want to ask some questions regarding this

**Comment by abdullah-afzal** [comment]
[Here](https://docs.medusajs.com/resources/references/payment/provider) you can check complete guide for creating a new payment provider.
